File: 929_Unique_Email_Addresses/sol.py
def process(mail: str):
    dogsign = mail.find('@');
    local, domain = mail[:dogsign], mail[dogsign+1:]
    local=local[:local.find('+')].replace(".", "")
    return "".join([local, domain])

class Solution:
    def numUniqueEmails(self, emails: 'List[str]') -> 'int':
        return len(set([process(e) for e in emails]))

File: 929_Unique_Email_Addresses/sol.cpp
string process(string& email){
    auto dogsign = email.find("@");
    string local = email.substr(0, dogsign);
    local = local.substr(0, local.find("+"));
    local.erase(std::remove(local.begin(), local.end(), '.'), local.end());
    string domain = email.substr(dogsign);
    return local+domain;
}

class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        std::transform(emails.begin(), emails.end(), emails.begin(), process);
        std::set<string> s(emails.begin(), emails.end());
        return s.size();
    }
};
